Flames/flames_drv.py

```diff
@@ -184,24 +171,39 @@ def run(self):
                 msgObj = json.loads(cmd)
                 type = msgObj["cmdType"]
                 logger.debug("message is {}".format(msgObj))
-                if type == "flameEffectStart":
-                    # figure out firing sequence associated with the name
-                    sequence = pattern_manager.getPattern(msgObj["name"])
-                    if self.checkSequence(msgObj):
-                        # set up poofer events
-                        pass
-                    event_manager.postEvent({"msgType":"sequence_start", "id":msgObj["name"]})
-                # else - whatever other type of event you want to process ...
+                if type == "stop":
+                    self.stopAll()
+
+                elif type == "resume":
+                    self.resumeAll()
+
+                elif type == "pooferDisable":
+                    # TODO: does this need to persist?
+                    self.disablePoofer(msgObj)
+
+                elif not isStopped:
+                    if type == "flameEffectStop":
+                        self.stopFlameEffect(msgObj)
+
+                    elif type == "pooferEnable":
+                        self.enablePoofer(msgObj)
+
+                    elif type == "flameEffectStart":
+                        self.startFlameEffect(msgObj)
+                        # else - whatever other type of event you want to process ...
+                    elif type == "flameEffectStop":
+                        self.stopFlameEffect(msgObj)
+
             except Queue.Empty:
                 # this is just a timeout - completely expected. Run the loop
                 pass
             except Exception:
-                logger.exception("Unexpected exception processing command queue!") 
-                
-                
-                
+                logger.exception("Unexpected exception processing command queue!")
+
+
+
     ## sanity and limit checks on an event from firingSequence ##
-    # CSW - suggest just returning True or False, and logging the reason, rather than 
+    # CSW - suggest just returning True or False, and logging the reason, rather than
     # returning the reason as a string. The upper level code really doesn't care why
     # the call failed.
     def checkEvent(self, event):
```

Comment:
Either get rid of this or rewrite it so the event syntax matches what we're using. It's not critical to check the event structure - this is a very limited application, and one can (mostly) assume that people aren't going to do anything wildly incorrect/stupid. (Hmm. As I'm thinking about that, the one wildly stupid thing that comes to mind is having an event that fires way, way after the other events. You might want to apply some discipline, such as 'no flame sequence can run for over a minute')